### PR TITLE
[TKR-275] Add Geist on Fantom

### PR DESCRIPTION
**Pull request by cece-z**
## Description

Add Geist on Fantom

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/asset-swapper/src/utils/market_operation_utils/geist_utils.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    const gTokenToUnderlyingToken = new Map<string, string>([
+        ['0x39b3bd37208cbade74d0fcbdbb12d606295b430a', '0x21be370d5312f44cb42ce377bc9b8a0cef1a4c83'], // gFTM => WFTM
+        ['0x940f41f0ec9ba1a34cf001cc03347ac092f5f6b5', '0x049d68029688eAbF473097a2fC38ef61633A3C7A'], // gFUSDT => fUSDT
+        ['0x07e6332dd090d287d3489245038daf987955dcfb', '0x8D11eC38a3EB5E956B052f67Da8Bdc9bef8Abf3E'], // gDAI => DAI
+        ['0xe578c856933d8e1082740bf7661e379aa2a30b26', '0x04068DA6C83AFCFA0e13ba15A6696662335D5B75'], // gUSDC => USDC
+        ['0x25c130b2624cf12a4ea30143ef50c5d68cefa22f', '0x74b23882a30290451A17c44f4F05243b6b58C76d'], // gETH => ETH
+        ['0x38aca5484b8603373acc6961ecd57a6a594510a3', '0x321162Cd933E2Be498Cd2267a90534A804051b11'], // gWBTC => BTC
+        ['0x690754a168b022331caa2467207c61919b3f8a98', '0x1E4F97b9f9F913c46F1632781732927B9019C68b'], // gCRV => CRV
+        ['0xc664fc7b8487a3e10824cda768c1d239f2403bbe', '0x82f0B8B456c1A451378467398982d4834b6829c1'], // gMIM => MIM
+    ]);
```
Comment (kimpers) [inline packages/asset-swapper/src/utils/market_operation_utils/geist_utils.ts:17]: SUPER NIT: If you move this outside the function it only gets initialized once, whereas now it's initialized on every call

File: packages/protocol-utils/src/transformer_utils.ts [hunk 2]
```diff
@@ -134,6 +134,7 @@ export enum BridgeProtocol {
     Clipper, // Not used: Clipper is now using PLP interface
     AaveV2,
     Compound,
+    Geist,
```
Comment (kimpers) [inline packages/protocol-utils/src/transformer_utils.ts:137]: This is a 1:1 Aave V2 fork right? Then you don't need to add anything here.

This list needs to match [BridgeProtocols.sol](https://github.com/0xProject/protocol/blob/development/contracts/zero-ex/contracts/src/transformers/bridges/BridgeProtocols.sol#L26-L55) so we only add unique integrations that require a deployment :)

File: packages/asset-swapper/src/utils/market_operation_utils/orders.ts [hunk 3]
```diff
@@ -202,6 +203,8 @@ export function getErc20BridgeSourceToBridgeSource(source: ERC20BridgeSource): s
             return encodeBridgeSourceId(BridgeProtocol.AaveV2, 'AaveV2');
         case ERC20BridgeSource.Compound:
             return encodeBridgeSourceId(BridgeProtocol.Compound, 'Compound');
+        case ERC20BridgeSource.Geist:
+            return encodeBridgeSourceId(BridgeProtocol.Geist, 'Geist');
```
Comment (kimpers) [inline packages/asset-swapper/src/utils/market_operation_utils/orders.ts:207]: If this is a 1:1 fork you can use `BridgeProtocol.AaveV2` just like we have done with the gazillion Uniswap V2 forks above 😄 
```suggestion
            return encodeBridgeSourceId(BridgeProtocol.AaveV2, 'Geist');
```

**Review by kimpers — CHANGES_REQUESTED** [review]
Awesome work on this! Just some minor comments and nits (feel free to ignore) 🙏

File: packages/asset-swapper/src/utils/market_operation_utils/constants.ts [hunk 4]
```diff
@@ -1996,6 +2015,13 @@ export const COMPONENT_POOLS_BY_CHAIN_ID = valueByChainId(
     },
 );
 
+export const GEIST_INFO_ADDRESS_BY_CHAIN_ID = valueByChainId<string>(
+    {
+        [ChainId.Fantom]: '0xd8321AA83Fb0a4ECd6348D4577431310A6E0814d',
```
Comment (kimpers) [inline packages/asset-swapper/src/utils/market_operation_utils/constants.ts:2020]: NIT: we try to lowercase addresses

**Review by dekz — APPROVED** [review]

**Pull request merged**